File: XText55.cpp
#include "pt4.h"
#include <fstream>
#include <algorithm>
#include <string>
using namespace std;

void Solve()
{
	Task("XText55");
	string s1, s2;
	pt >> s1 >> s2;
	ifstream f1(s1);
	ofstream f2(s2, ios::binary);
	string s;
	char c;
	while(f1){
		c = f1.get();
		if(isalnum(c) || ispunct(c) || c == ' '){
				s = s + c;
		}
	}
	sort(s.begin(), s.end());
	for(int i = 1; i < s.length();){
		if(s[i] == s[i -1]){
			s.erase(i,1);
		}else
			i++;
	}
	f2 << s;
	f1.close();
	f2.close();
}

File: XText55.cs
// File: "XText55"
using PT4;
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XText55");
            StreamReader r = new StreamReader(GetString());
            FileStream f = File.OpenWrite(GetString());
            BinaryWriter br = new BinaryWriter(f);

            SortedSet<char> symbolist = new SortedSet<char>();
            while (!r.EndOfStream)
            {
                char c = ((char)r.Read());
                if (!symbolist.Contains(c))
                {
                    symbolist.Add(c);
                }
            }
            foreach (char c in symbolist)
            {
                if ((int)c != 10 && (int)c != 13)
                    br.Write(c);
            }
            r.Close();
            br.Close();
            f.Close();
        }
    }
}
